### ISSUE TITLE
[gcp_authn] How to configure authn cluster

### ISSUE THREAD

**Issue by a1lu** [envoyproxy/envoy#24269]
Hi,
I am trying to get the [GCP Authn Filter working](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/gcp_authn_filter) working with gRPC requests.
I copied the authn examples from the docs but I don't know how to configure the gcp_authn endpoint. The example uses `0.0.0.0:8000` like the fake service and envoy itself.

* In envoy logs I find this error related to gcp_authn
```
[2022-11-30 16:09:58.397][8][error][init] [source/extensions/filters/http/gcp_authn/gcp_authn_impl.cc:81] Response status is not OK, status: 503
```

* In the logs of the receiving service I can see the error 401 for my calls (as expected).
* I create a envoy container in google cloud run using the base image `envoyproxy/envoy-distroless:v1.23-latest`

My envoy.yaml:

```
static_resources:
  listeners:
  - name: listener_0
    address:
      socket_address: { address: 0.0.0.0, port_value: 80 }
    filter_chains:
    - filters:
      - name: envoy.filters.network.http_connection_manager
        typed_config:
          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
          codec_type: auto
          stat_prefix: ingress_http
          route_config:
            name: local_route
            virtual_hosts:
            - name: local_service
              domains: ["*"]
              routes:
              - match: { prefix: "/" }
                route:
                  cluster: my_service
                  auto_host_rewrite: true
                  max_stream_duration:
                    grpc_timeout_header_max: 0s
          http_filters:
          - name: envoy.filters.http.gcp_authn
            typed_config:
              "@type": type.googleapis.com/envoy.extensions.filters.http.gcp_authn.v3.GcpAuthnFilterConfig
              http_uri:
                uri: "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/identity?audience=[AUDIENCE]"
                cluster: "gcp_authn"
                timeout:
                  seconds: 10
          - name: envoy.filters.http.router
            typed_config:
              "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
  clusters:
  - name: my_service
    type: strict_dns
    connect_timeout: 20s
    typed_extension_protocol_options:
      envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
        "@type": type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
        explicit_http_config:
          http2_protocol_options:
            { }
    lb_policy: round_robin
    dns_refresh_rate: 90s
    metadata:
      typed_filter_metadata:
        envoy.filters.http.gcp_authn:
          "@type": type.googleapis.com/envoy.extensions.filters.http.gcp_authn.v3.Audience
          url: https://my_service-xyz.a.run.app
    load_assignment:
      cluster_name: my-service
      endpoints:
        - lb_endpoints:
            - endpoint:
                address:
                  socket_address:
                    address: my_service-xyz.a.run.app
                    port_value: 443
    dns_lookup_family: V4_ONLY
    transport_socket:
      name: envoy.transport_sockets.tls
      typed_config:
        "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
        common_tls_context:
          alpn_protocols: h2
          validation_context:
            trusted_ca:
              filename: /etc/ssl/certs/ca-certificates.crt
        sni: https://my_service-xyz.a.run.app
  - name: gcp_authn
    load_assignment:
      cluster_name: gcp_authn
      endpoints:
      - lb_endpoints:
        - endpoint:
            address:
              socket_address:
                address: ? what enter here ?
                port_value: ? what enter here ?
    typed_extension_protocol_options:
      envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
        "@type": type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
        explicit_http_config:
          http2_protocol_options:
            {}
```

Could you please tell me what the correct/expected endpoint is?

**Comment by tyxia** [comment]
@a1lu. This is the address for your authentication server.  Yes, the example in the doc is from fake server.

In the real production case where GCP metadata server is the authentication server, the config, for example `strict_dns` type,  could be:
`type: STRICT_DNS
      connect_timeout {
        seconds: 5
      }
      dns_lookup_family: V4_ONLY
      load_assignment {
        cluster_name: "GCP_METADATA_SERVER_CLUSTER"
        endpoints {
          lb_endpoints{
            endpoint {
              address {
                socket_address {
                  address: "metadata.google.internal"
                  port_value: 80
                }
              }
            }
          }
        }
      }`

**Comment by a1lu** [comment]
@tyxia 
Thanks for your response. 

Unfortunately I still can't reach my service.

In the envoy cloud run log I still see
```
[2022-12-02 08:11:15.961][8][error][init] [source/extensions/filters/http/gcp_authn/gcp_authn_impl.cc:81] Response status is not OK, status: 503
```

And my call returns unauthenticated.

My cluster config
```
  - name: gcp_authn
    type: STRICT_DNS
    dns_lookup_family: V4_ONLY
    load_assignment:
      cluster_name: gcp_authn
      endpoints:
      - lb_endpoints:
        - endpoint:
            address:
              socket_address:
                address: "metadata.google.internal"
                port_value: 80
    typed_extension_protocol_options:
      envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
        "@type": type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
        explicit_http_config:
          http2_protocol_options:
            {}
```

**Comment by a1lu** [comment]
By removing the part
```
    typed_extension_protocol_options:
      envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
        "@type": type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
        explicit_http_config:
          http2_protocol_options:
            {}
```
I could solve the message
```[2022-12-02 08:11:15.961][8][error][init] [source/extensions/filters/http/gcp_authn/gcp_authn_impl.cc:81] Response status is not OK, status: 503
```
Still not authenticated though. But now I can debug with an open service.

**Comment by a1lu** [comment]
It works now. Thanks a lot @tyxia 

After I removed the  `http2_protocol_options` I ran into https://github.com/envoyproxy/envoy/issues/22916. This was solved by using `envoyproxy/envoy-distroless:v1.24-latest` as base image

**Comment by tyxia** [comment]
> It works now. Thanks a lot @tyxia
> 
> After I removed the `http2_protocol_options` I ran into #22916. This was solved by using `envoyproxy/envoy-distroless:v1.24-latest` as base image

@a1lu  I am glad to help and it is working now. Thanks for your feedback! 

I do agree the original configuration from integration test with fake server might be a bit confusing. I am going to update the doc.

**Comment by a1lu** [comment]
@tyxia 
Do you have any experience with the correct service account roles that the envoy service needs to be able to use the received ID token?
